frontend/src/app/components/wp-single-view-tabs/additional-tab/additional-tab.component.ts

```diff
@@ -0,0 +1,95 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See docs/COPYRIGHT.rdoc for more details.
+//++
+
+import {Transition} from '@uirouter/core';
+import {AfterViewInit, Component, ComponentFactoryResolver, Input, OnInit, ViewChild, ViewContainerRef} from '@angular/core';
+import {I18nService} from 'core-app/modules/common/i18n/i18n.service';
+import {WorkPackageResource} from 'core-app/modules/hal/resources/work-package-resource';
+import {HookService} from 'core-app/modules/plugins/hook-service';
+import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
+import {APIV3Service} from "core-app/modules/apiv3/api-v3.service";
+import { TabComponent } from './tab.component';
+import { Tab } from './tab';
+
+@Component({
+  templateUrl: './additional-tab.html',
+  selector: 'wp-additional-tab',
+})
+export class WorkPackageAdditionalTabComponent extends UntilDestroyedMixin implements OnInit, AfterViewInit {
+  @Input() public workPackageId?:string;
+  @ViewChild('additionalTabContent', { read: ViewContainerRef }) additionalTabContent: ViewContainerRef;
+
+  public additionalTab:Tab|undefined;
+  public workPackage:WorkPackageResource;
+
+  public constructor(readonly I18n:I18nService,
+                     readonly $transition:Transition,
+                     readonly apiV3Service:APIV3Service,
+                     readonly HookService:HookService,
+                     private componentFactoryResolver: ComponentFactoryResolver) {
+    super();
+  }
+
+  findTab() {
+    const additionalTabIdentifier = this.$transition.params('to').additionalTabIdentifier;
+    const registeredAdditionalTabs = this.HookService.call('workPackageAdditionalTabs');
+    const additionalTabs = this.workPackage.additionalTabs(registeredAdditionalTabs);
+    this.additionalTab = _.find(additionalTabs, ({identifier: id}) => id === additionalTabIdentifier);
+  }
+
+  ngOnInit() {
+    const wpId = this.workPackageId || this.$transition.params('to').workPackageId;
+    this
+      .apiV3Service
+      .work_packages
+      .id(wpId)
+      .requireAndStream()
+      .pipe(
+        this.untilDestroyed()
+      )
+      .subscribe((wp) => {
+        this.workPackageId = wp.id!;
+        this.workPackage = wp;
+        this.findTab();
+      });
+  }
+
+  ngAfterViewInit() {
+    setTimeout(() => { // TODO: setTimeout is a crazy hack, but without it `additionalTabContent` is undefined. Don't know why, according to the internet it should work just fine. Could need help debugging this.
```

Comment:
this drove me crazy - I can not access the `<ng-container>` defined in frontend/src/app/components/wp-single-view-tabs/additional-tab/additional-tab.html in `ngAfterViewInit()` although the angular docs say everything should be rendered already. It works consistently after a timeout, though. Is there a better way to do this? @oliverguenther  maybe we can pair-debug this.

Reply:
You have a ViewChild within an `ngIf` that is only determined after `ngOnInit`. This is a common confusion with how ViewChild will work in this scenario. You can work around it by using a setter on the ViewChild like in https://stackoverflow.com/a/41095677/420614.

But I would like to suggest a different way of mounting the dynamic component that doesn't need the whole shebang of ComponentFactoryResolver and manually instantiating the component. We can discuss this today afternooon if you have time